src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp

```diff
@@ -97,213 +92,409 @@ Position const SpawnBat[6] =
-                    break;
-                case EVENT_SPAWN_FLYING_BATS:
-                    Talk(SAY_CALL_RIDERS);
-                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
-                        if (Creature* flyingBat = me->SummonCreature(NPC_FRENZIED_BAT, target->GetPositionX(), target->GetPositionY(), target->GetPositionZ() + 15.0f, 0.0f, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
-                            flyingBat->AI()->DoZoneInCombat();
-                    events.ScheduleEvent(EVENT_SPAWN_FLYING_BATS, 10s, 15s, PHASE_TWO);
-                    break;
-                default:
-                    break;
-            }
+            Reset();
         }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        // ensures that the scheduler gets updated even out of combat
+        scheduler.Update(diff);
+
+        BossAI::UpdateAI(diff);
+    }
 
-        DoMeleeAttackIfReady();
+    void JustDied(Unit* killer) override
+    {
+        BossAI::JustDied(killer);
+        Talk(SAY_DEATH);
     }
 };
 
-// Flying Bat
-struct npc_batrider : public ScriptedAI
+// Gurubashi Bat Rider (14750) - trash and boss summon are same creature ID
+struct npc_batrider : public CreatureAI
 {
-    npc_batrider(Creature* creature) : ScriptedAI(creature)
+    BatRiderMode _mode; // the version of this creature (trash or boss)
+    TaskScheduler _scheduler;
+
+    npc_batrider(Creature* creature) : CreatureAI(creature)
     {
-        _scheduler.SetValidator([this]
+        // if this is a summon of Jeklik, it is in boss mode
+        if
+        (
+            me->GetEntry() == NPC_BATRIDER &&
+            me->IsSummon() &&
+            me->ToTempSummon() &&
+            me->ToTempSummon()->GetSummoner() &&
+            me->ToTempSummon()->GetSummoner()->GetEntry() == NPC_PRIESTESS_JEKLIK
+        )
         {
-            return !me->HasUnitState(UNIT_STATE_CASTING);
-        });
+            LOG_DEBUG("scripts.ai", "npc_batrider::constructor: BATRIDER_MODE_BOSS");
+            _mode = BATRIDER_MODE_BOSS;
+
+            // make the bat rider unattackable
+            me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+            me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+            me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
+
+            // keep the bat from attacking players directly
+            me->SetReactState(REACT_PASSIVE);
+
+            // make the bat rider move the correct speed
+            me->SetSpeed(MOVE_WALK, 5.0f, true);
+
+            // start the flight loop
+            me->SetCanFly(true);
+            me->SetDisableGravity(true);
+            me->GetMotionMaster()->MoveSplinePath(PATH_BATRIDER_LOOP);
+        }
+        // otherwise, trash mode
+        else
+        {
+            LOG_DEBUG("scripts.ai", "npc_batrider::constructor: BATRIDER_MODE_TRASH");
+            me->SetReactState(REACT_DEFENSIVE);
+            _mode = BATRIDER_MODE_TRASH;
```

Comment:
Personal grievance: maybe move this `_mode =` directly under the log as it is in the boss summon block for consistency.

Reply:
Done.